visualization/dashboard/src/MLIDashboard/Controls/CohortEditor/CohortEditor.tsx

```diff
@@ -372,14 +417,59 @@ export class CohortEditor extends React.PureComponent<ICohortEditorProps, ICohor
     }
 
     private setFilterLabel(filter: IFilter): React.ReactNode {
-        //TODO: change the function unwrap neatly 
+        const selectedFilter = this.props.jointDataset.metaDict[filter.column];
         let label = "";
-        const space = " ";
-        label = this.props.jointDataset.metaDict[filter.column].abbridgedLabel + space
-        label += filter.method + space
-        label += filter.arg
+        label = selectedFilter.abbridgedLabel
+
+        const filterMethod = this.getFilterMethodLabel(filter.method)
+            label += filterMethod
+        if (selectedFilter.isCategorical) {
+            let selectedValues = [];
+            let filterArgs = filter.arg as number[];
+            filterArgs.forEach((element) => {
+                selectedValues.push(selectedFilter.sortedCategoricalValues[element])
+            }
+            );
+            label += selectedValues;
+        }
+        else {
+            label += filter.arg
```

Comment:
It seems the tooltip follows a structured pattern, this is often a good place for the template literal mentioned above (made using backticks and escaped with ${} to include javascript). That would allow you to set the structure like `${abbridgedLabel} ${localization.formatString(filterStringDict[filter.method], stringifiedArgs)}`;

Reply:
Added template literal however this method will need some simplification.